File: android/src/main/java/au/com/crazybean/mobilex/kurir/modules/find/impl/FindModule.kt
package au.com.crazybean.mobilex.kurir.modules.find.impl

import au.com.crazybean.mobilex.kurir.modules.find.FindDelegate
import au.com.crazybean.mobilex.kurir.modules.find.FindView
import au.com.crazybean.mobilex.kurir.modules.find.FindViewModel
import org.koin.androidx.viewmodel.dsl.viewModel
import org.koin.dsl.module

val findModule = module {
    // ViewModel
    viewModel {
        FindViewModel()
    }

    // Delegate
    factory { (view: FindView?) ->
        FindDelegate(view, get())
    }
}

File: ios/Kurir/modules/find/impl/FindModule.swift
//
//  FindModule.swift
//  Kurir
//
//  Created by Loren on 19/9/19.
//  Copyright © 2019 Crazybean Studio. All rights reserved.
//

import Mobilex

class FindModule: Module {
    override func inject() {
        // ViewModel
        factory(FindViewModel.self) { _ in
            FindViewModel()
        }
        
        // Delegate
        factory(FindDelegate.self) { (r, view: FindView) -> FindDelegate in
            FindDelegate(view: view, viewModel: r.resolve(FindViewModel.self)!)
        }
    }
}
